context/context.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2024, Ava Labs, Inv. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package context
+
+import (
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Metrics interface {
+	// Register adds the outputs of [gatherer] to the results of future calls to
+	// Gather with the provided [name] added to the metrics.
+	Register(name string, gatherer prometheus.Gatherer) error
+}
+
+type Context struct {
```

Comment:
I generally dislike these `Context` patterns (we do this in avalanchego... but I still dislike `snow.Context` because I feel like it makes it too easy to pass around global state/dependencies that a type really doesn't care about. It's also unclear to me why some dependencies in `Initialize` exist/don't exist in `snow.Context`, like the database... I feel like we should be consistent). I would probably prefer us just passing dependencies around individually as parameters.

Reply:
narrowing to just using this for config. This does mean we go back to using AvalancheGo's snow context for creating metrics, passing the log in some places, and add the tracer to `ChainInput`